src/containers/EditListingPage/EditListingWizard/EditListingDetailsPanel/EditListingDetailsPanel.js

```diff
@@ -271,6 +314,16 @@ const EditListingDetailsPanel = props => {
               ...rest
             } = values;
 
+            // Category levels are dynamic, i.e. in theory you can always nest
+            // another subcategory under a subcategory, therefore  we loop through them
+            const categories = {};
+            let level = 1;
+            while (typeof values[`${CATEGORY_PREFIX}${level}`] !== 'undefined') {
+              const currentCategoryKey = `${CATEGORY_PREFIX}${level}`;
+              categories[currentCategoryKey] = values[currentCategoryKey];
+              level++;
+            }
```

Comment:
How about adding a helper function like this somewhere:
```js
const pickCategoryFields = (formValues, level) => {
  const currentCategoryKey = `${CATEGORY_PREFIX}${level}`;
  const currentCategoryValue = formValues[currentCategoryKey];
  const isCategoryLevelSet = typeof currentCategoryValue !== 'undefined';

  return isCategoryLevelSet 
    ? { [currentCategoryKey]: currentCategoryValue, ...updateCategoryFields(formValues, ++level) }
    : {};
}

```
Then in this submit function, you could just call it like this:
```js
// Category levels are dynamic, i.e. in theory you can always nest
// another subcategory under a subcategory, therefore we loop through them
const categories = pickCategoryFields(values, 1);
```